lib/beaker-pe/install/pe_utils.rb

```diff
@@ -325,6 +325,26 @@ def fetch_pe_on_mac(host, opts)
           end
         end
 
+        # Determine the build package to download on a sles-11 (Intel) host, install that package onto the host.
+        # Assumed file name format: puppet-agent-7.29.1.26.gf344eeefa-1.sles11.x86_64.rpm.
+        # This method should be called after puppet is installed on the master since it relies on the master
+        # telling it the puppet agent version to form the download URL.
+        # @param [Host] host The sles-11 host to download and install the package on.
+        # @param  [Hash{Symbol=>Symbol, String}] opts The options
+        # @api private
+        def install_rpm_on_sles11_host(host, opts)
+          puppet_agent_ver = get_puppet_agent_version(host, opts)
+          # Since sles11 builds are not available in PE, download from agent-downloads.
+          agent_downloads_url = "http://agent-downloads.delivery.puppetlabs.net/puppet-agent"
+          master_aio_version = puppet_fact(master, 'aio_agent_build')
+          # The first digit of the puppet agent version indicates the stream.
+          stream = puppet_agent_ver[0]
+          path = "#{agent_downloads_url}/#{puppet_agent_ver}/repos/sles/11/puppet#{stream}/x86_64"
```

Comment:
The `stream` is already available in the `opts` hash passed into this method as `opts[:puppet_collection]`. Note it's of the form `puppet7`, see

 https://github.com/puppetlabs/beaker-pe/blob/6f7743054671033a5a22348f1b6d2d4529b02349/lib/beaker-pe/install/pe_utils.rb#L920

Reply:
I noticed sometimes the opts[:puppet_collection] was nil while testing locally. To be sure that stream is not nil because of this added the `|| "puppet#{puppet_agent_ver[0]}"` part